os/drivers/compression/compress.c

```diff
@@ -87,9 +95,12 @@ static ssize_t comp_write(FAR struct file *filep, FAR const char *buffer, size_t
 static int comp_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 {
 	int ret = -ENOSYS;
+	unsigned int size;
+	off_t filelen;
 	struct compress_header *comp_info = (struct compress_header *)arg;
+	struct file_decomp_data_s *data;
 
-	if (comp_info == NULL && cmd != COMPIOC_GET_COMP_TYPE) {
+	if (comp_info == NULL && !((cmd == COMPIOC_GET_COMP_TYPE) || (cmd==COMPIOC_GET_COMP_HEADER) || (cmd ==COMPIOC_UNINIT))) {
```

Comment:
This condition seems wrong and unnecessary. Please change it as below.

```
if (comp_info == NULL) {
    return -EINVAL;
}
```

Also, add a default case in the case statement and return EINVAL from there.